src/include/miopen/backend_api_builders.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <miopen/miopen.h>
+#include <miopen/tensor.hpp>
+#include <memory>
+#include <vector>
+
+namespace miopen {
+
+namespace backend_api {
```

Comment:
optional suggestions: 
rename `backend_api` to `backend` or `backendapi`.

Reply:
Agree. In cudnn these APIs are called Graph API
![image](https://github.com/ROCm/MIOpen/assets/102019268/b6b66dc8-b1cd-4d8a-8545-3d90ec3df8cf)
Is it better to name it `graphapi`?

Reply:
yeah, I personally like `graphapi` and I don't like the name `backend` but seems like cudnn also chose wrong and named every enum and api with `BackendXYZ`. So, I say let's rename our internal `namespace` to `graphapi` and then we can think about what to do with the word `Backend` in the public API and enums. This entire API feels like a series of bad decisions ;).

Reply:
What about filenames? I suggest to following renames:
backend_api.hpp -> graphapi.hpp
backend_api.cpp -> graphapi.cpp
backend_tensor.hpp -> graphapi_tensor.hpp
backend_tensor.cpp -> graphapi_tensor.hpp

Reply:
I'm OK with that. @JehandadKhan : please provide your input on this.